cdap-proto/src/main/java/co/cask/cdap/proto/provisioner/ProvisionerDetail.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.proto.provisioner;
+
+import com.google.gson.JsonElement;
+import com.google.gson.annotations.SerializedName;
+
+import java.util.Objects;
+
+/**
+ * A class which contains the config of a provisioner
+ */
+public class ProvisionerDetail {
+  private final String name;
+  private final String description;
+  @SerializedName("configuration-groups")
+  private final JsonElement configurationGroups;
```

Comment:
does gson preserve the structure if you change this to `List<Object>`? We know it's supposed to be a List, and trying to avoid using gson specific classes for types.